bin/install_uninstall.ml

```diff
@@ -119,6 +143,56 @@ let file_operations ~dry_run : (module FILE_OPERATIONS) =
   else
     (module File_ops_real)
 
+module Git_version = struct
+  let git = lazy (
+    match Bin.which ~path:(Env.path Env.initial) "git" with
+    | Some x -> x
+    | None -> Utils.program_not_found "git" ~loc:None)
+
+  let get =
+    Memo.create
+      "git-describe"
+      ~doc:"Run git describe in the following directory"
+      ~input:(module Path)
+      ~output:(Simple (module struct
+                 type t = string option
+                 let to_sexp = Sexp.Encoder.(option string)
+               end))
+      ~visibility:(Public Path_dune_lang.decode)
+      Async
+      None
+
+  let () =
+    let f path =
+      if Path.is_directory (Path.relative path ".git") then
+        let env = Env.initial in
+        Dune.Process.run_capture Strict (Lazy.force git)
+          ["describe"; "--always"; "--dirty"] ~env
+        >>| fun s ->
+        Option.some (String.trim s)
+      else
+        match Path.parent path with
+        | None -> Fiber.return None
+        | Some path -> Memo.exec get path
+    in
+    Memo.set_impl get f
+
+  let get_for_file_to_install path =
+    let path =
+      (* Follow one level of symlink as install paths are always the
+         symlinks in [_build/install]]. *)
+      try
+        let s = Unix.readlink (Path.to_string path) in
+        Path.relative (Path.parent_exn path) s
+      with _ ->
```

Comment:
Which exception is this designed to catch? The one raised by `Unix.readlink`? If that's teh case, I would simply surround Unix.readlink with that.

Reply:
indeed, I'll do that